src/app/containers/ArticleMain/index.test.jsx

```diff
@@ -61,23 +71,42 @@ it('should use headline for meta description if summary does not exist', async (
   ).toEqual('Article Headline for SEO');
 });
 
-shouldMatchSnapshot(
-  'should render a news article correctly',
-  <Context service="news">
-    <ArticleMain articleData={articleDataNews} />
-  </Context>,
-);
+it('should render a news article correctly', async () => {
+  fetch.mockResponse(JSON.stringify(newsMostReadData));
 
-shouldMatchSnapshot(
-  'should render a persian article correctly',
-  <Context service="persian">
-    <ArticleMain articleData={articleDataPersian} />
-  </Context>,
-);
+  const { container } = render(
+    <Context service="news">
+      <ArticleMain articleData={articleDataNews} />
+    </Context>,
+  );
 
-shouldMatchSnapshot(
-  'should render a pidgin article correctly (with navigation)',
-  <Context service="pidgin">
-    <ArticleMain articleData={articleDataPidgin} />
-  </Context>,
-);
+  expect(container).toMatchSnapshot();
+});
+
+it('should render a rtl article (persian) with most read correctly', async () => {
+  fetch.mockResponse(JSON.stringify(persianMostReadData));
+
+  const { container } = render(
+    <Context service="persian">
+      <ArticleMain articleData={articleDataPersian} />
+    </Context>,
+  );
+
+  await waitForElement(() => container.querySelector('#Most-Read'));
+
+  expect(container).toMatchSnapshot();
+});
+
+it('should render a ltr article (pidgin) with most read correctly', async () => {
+  fetch.mockResponse(JSON.stringify(pidginMostReadData));
+
+  const { container } = render(
+    <Context service="pidgin">
+      <ArticleMain articleData={articleDataPidgin} />
+    </Context>,
+  );
+
+  await waitForElement(() => container.querySelector('#Most-Read'));
+
+  expect(container).toMatchSnapshot();
+});
```

Comment:
It's good to add these tests at the unit level, so that we don't need to add cypress ones for the page to check for Most Read on articles!